tests/test_computer.py

```diff
@@ -130,3 +130,73 @@ def test_dynamic_info(firecrest_config, monkeypatch, tmpdir: Path):
     # note: user cannot enter negative numbers anyways, click raise as this shoule be float not str
     result = _dynamic_info_direct_size(ctx, None, 10)
     assert result == 10
+
+
+def test_dynamic_info_firecrest_version(
+    firecrest_config, monkeypatch, tmpdir: Path, capsys
+):
+
+    from unittest.mock import patch
+
+    from click import Abort
+
+    from aiida_firecrest.transport import _dynamic_info_firecrest_version
+    from conftest import MockFirecrest
+
+    ctx = Mock()
+    ctx.params = {
+        "url": f"{firecrest_config.url}",
+        "token_uri": f"{firecrest_config.token_uri}",
+        "client_id": f"{firecrest_config.client_id}",
+        "client_secret": f"{firecrest_config.client_secret}",
+        "compute_resource": f"{firecrest_config.compute_resource}",
+        "temp_directory": f"{firecrest_config.temp_directory}",
+        "api_version": f"{firecrest_config.api_version}",
+    }
+
+    # should catch FIRECREST_VERSION if value is not provided
+    result = _dynamic_info_firecrest_version(ctx, None, "None")
+    assert isinstance(result, str)
+
+    # should use the value if provided
+    result = _dynamic_info_firecrest_version(ctx, None, "10.10.10")
+    assert result == "10.10.10"
+
+    # raise BadParameter if the version is not supported
+    with pytest.raises(BadParameter) as exc_info:
+        result = _dynamic_info_firecrest_version(ctx, None, "0.0.0")
+    assert "FirecREST api version 0.0.0 is not supported" in str(exc_info.value)
```

Comment:
pytest has an argument to match the error message
```suggestion
    with pytest.raises(BadParameter, match=r".*FirecREST api version 0.0.0 is not supported.*")):
       _dynamic_info_firecrest_version(ctx, None, "0.0.0")
```